pkg/core/middleware/cors.go

```diff
@@ -25,22 +27,32 @@ type CORSMiddleware struct {
 
 func (cors CORSMiddleware) Handle(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		requestMethod := r.Method
-		corsMethod := r.Header.Get("Access-Control-Request-Method")
-		corsHeaders := r.Header.Get("Access-Control-Request-Headers")
-
 		tConfig := config.GetTenantConfig(r)
-		w.Header().Set("Access-Control-Allow-Origin", tConfig.CORSHost)
-		w.Header().Set("Access-Control-Allow-Credentials", "true")
-
-		if corsMethod != "" {
-			w.Header().Set("Access-Control-Allow-Methods", corsMethod)
+		matcher, err := originmatcher.Parse(tConfig.CORSHost)
+		if err != nil {
+			panic(fmt.Errorf("Unable to parse CORS host config: %v", err.Error()))
```

Comment:
Let's not panic if parsing failed. Treat this case as never matched. So

```golang
if err != nil && origin != "" && matcher.MatchOrigin(origin) {
```

Reply:
This would be a silent configuration failure, which app developer may have a hard time to figure out why it's not returning correct CORS headers.

If panic is undesirable, I suppose the CORS host config would be validated before committing into DB?

Reply:
Currently we have no ways for the developer to config CORS host. The observed existing values are `""` and `"*"`. When we allow the developer to edit it, we definitely will use originmatcher to parse it once before storing in database.

Reply:
I just discovered a bug in the library. I have fixed it

https://github.com/iawaknahc/originmatcher/commits/master

Please update.